nRF51ServiceDiscovery.cpp

```diff
@@ -227,10 +227,9 @@ nRF51ServiceDiscovery::CharUUIDDiscoveryQueue::triggerFirst(void)
             .uuid = BLE_UUID_CHARACTERISTIC,
             .type = BLE_UUID_TYPE_BLE,
         };
-        ble_gattc_handle_range_t handleRange = {
-            .start_handle = parentDiscoveryObject->characteristics[charIndex].getDeclHandle(),
-            .end_handle   = parentDiscoveryObject->characteristics[charIndex].getDeclHandle() + 1,
-        };
+        ble_gattc_handle_range_t handleRange = { };
+        handleRange.start_handle = parentDiscoveryObject->characteristics[charIndex].getDeclHandle();
+        handleRange.end_handle   = parentDiscoveryObject->characteristics[charIndex].getDeclHandle() + 1;
```

Comment:
This one was weird.

So apparently, and in full conformance to ISO-C/C++ rules, the `GattAttribute::Handle_t` (which is a `uint16_t` type) is, when you add `1`, promoted to an `int`... and then you get an error because you are not allowed to narrow in a designated initializer.

You can cast all you want... setting `.start_handle` works fine, but adding anything to `.end_handle` upcasts to an `int` and you can't stop it.

Oddly enough, if you change the initializer syntax, the widening/narrowing does not happen.

Although this is a GCC-ism, is is perfectly valid and legal and annoying.